src/compiler/transformers/module/system.ts

```diff
@@ -120,6 +121,8 @@ namespace ts {
             );
 
             // Write the call to `System.register`
+            // Reset the source-file's nodeFlag to not emit helpers because we already set one in the body
+            // So the helper will be emit at the correct position instead of at the top of the source-file

```

Comment:
Consider creating a variable named `updatedNodeFlags` to keep this comment and the new flags nearby each other.